src/main/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandler.java

```diff
@@ -70,15 +77,19 @@ protected void onInit() throws Exception {
 
 	@Override
 	protected void handleMessageInternal(final Message<?> message) throws Exception {
-		String topic = this.topicExpression != null
-				? this.topicExpression.getValue(this.evaluationContext, message, String.class)
+		String topic = this.topicExpression != null ?
+				this.topicExpression.getValue(this.evaluationContext, message, String.class)
 				: message.getHeaders().get(KafkaHeaders.TOPIC, String.class);
 
+		Integer partitionId = this.partitionExpression != null ?
```

Comment:
I just followed the existing pattern :). However, I think the current form is, in a way, better, since it avoids a SpEL evaluation if an expression is not actually set. I think that getting the header from the map is generally more efficient.